examples/lightgbm_link_baseline.py

```diff
@@ -0,0 +1,314 @@
+col_to_stype[target_col_name] = torch_frame.categorical
+
+# Prepare train/val dataset for lightGBM model training. For each lhs
+# entity, their corresponding rhs entities are used as positive label.
+# The same number of random rhs entities are sampled as negative label.
+# lightGBM will train and eval on this binary classification task.
+left_entity = list(train_table.fkey_col_to_pkey_table.keys())[0]
+right_entity = list(train_table.fkey_col_to_pkey_table.keys())[1]
+for split, table in [
+    ("train", train_table),
+    ("val", val_table),
+]:
+    src_entity_df = src_entity_df.astype(
+        {src_entity_table.pkey_col: table.df[left_entity].dtype}
+    )
+
+    dst_entity_df = dst_entity_df.astype(
+        {dst_entity_table.pkey_col: table.df[right_entity].dtype}
+    )
+
+    # Left join train table and entity table
+    df = table.df.merge(
+        src_entity_df,
+        how="left",
+        left_on=left_entity,
+        right_on=src_entity_table.pkey_col,
+    )
+
+    # Transform the mapping between one lhs entity with a list of rhs entities
+    # to lhs entity, rhs entity pairs
+    df = df.explode(right_entity)
+
+    # Add a target col indicating there is a link between lhs and rhs entities
+    df[target_col_name] = 1
+
+    # Create a negative sampling df, containing lhs and rhs entities pairs,
+    # such that there are no links between them.
+    negative_sample_df_columns = list(df.columns)
+    negative_sample_df_columns.remove(right_entity)
+    negative_samples_df = df[negative_sample_df_columns]
+    negative_samples_df[right_entity] = np.random.choice(
+        dst_entity_df[dst_entity_table.pkey_col], size=len(negative_samples_df)
+    )
+    negative_samples_df[target_col_name] = 0
+
+    # Constructing a dataframe containing the same number of positive and
+    # negative links and
+    df = pd.concat([df, negative_samples_df], ignore_index=True)
+    df = pd.merge(
+        df,
+        dst_entity_df,
+        how="left",
+        left_on=right_entity,
+        right_on=dst_entity_table.pkey_col,
+    )
+    dfs[split] = df
+
+# Prepare test dataset for lightGBM model evalution
+test_df_column_names = list(test_table.df.columns)
+test_df_column_names.remove(right_entity)
+test_df = test_table.df[test_df_column_names]
+
+# Per each lhs entity, collect all past linked rhs entities
+trainval_table_df = pd.concat([train_table.df, val_table.df], axis=0)
+trainval_table_df.drop(columns=[train_table.time_col], inplace=True)
+
+
+def dst_entities_aggr(dst_entities):
+    r"concatenate and deduplicate rhs entities"
+    concatenated = [item for sublist in dst_entities for item in sublist]
+    return list(set(concatenated))
+
+
+grouped_deduped_trainval = (
+    trainval_table_df.groupby(left_entity)[right_entity]
+    .apply(dst_entities_aggr)
+    .reset_index()
+)
+test_df = pd.merge(test_df, grouped_deduped_trainval, how="left", on=left_entity)
+
+# collect the most popular rhs entities
+all_rhs_entities = [
```

Comment:
still a bunch of rhs/lhs terminologies to remove

Reply:
Fixing it right now